Core/Src/system.c

```diff
@@ -6,35 +6,50 @@
  */
 
 #include "system.h"
+#include "stdlib.h"
+
+//extern System my_sys;
+extern UART_HandleTypeDef huart3;
+
+
+extern TIM_HandleTypeDef htim2;
+extern TIM_HandleTypeDef htim3;
+extern TIM_HandleTypeDef htim4;
 
 
-extern System my_sys;
 void init_system(System *sys,
-		TIM_HandleTypeDef *h_timer1, TIM_HandleTypeDef *h_timer2,
 		GPIO_TypeDef *h_GPIO,
 		uint16_t LEDPin1, uint16_t LEDPin2 ,
-		uint16_t size_of_recieve_buffer) {
+		uint16_t size_of_receive_buffer) {
 
 	// fill my_sys with pointers and variables
-	sys->h_timer1 = h_timer1;
-	sys->h_timer2 = h_timer2;
+	sys->h_timer1 = &htim2;
```

Comment:
h_timerx's are used in the commandHandler for the LEDs to use either timer.  When a command comes in, it causes LEDx_timer to switch between h_timer1 and h_timer2 (or use neither), which need to be in the sys variable so it can get passed into the function.

Reply:
Ahhh yes that makes sense.  Very good!